lib/model/folder.go

```diff
@@ -182,16 +178,23 @@ func (f *folder) Serve(ctx context.Context) error {
 			return nil
 
 		case <-f.pullScheduled:
-			_, err = f.pull()
-
-		case <-f.pullFailTimer.C:
+			timeutil.StopAndDrain(pullFailTimer)
 			var success bool
+			t0 := time.Now()
 			success, err = f.pull()
 			if (err != nil || !success) && f.pullPause < 60*f.pullBasePause() {
 				// Back off from retrying to pull
 				f.pullPause *= 2
+
+				// Pulling failed, try again later.
+				delay := f.pullPause + time.Since(t0)
+				l.Infof("Folder %v isn't making sync progress - retrying in %v.", f.Description(), stringutil.NiceDurationString(delay))
+				timeutil.ResetTimer(pullFailTimer, delay)
 			}
```

Comment:
I don't think resetting the fail timer should be inside this condition. Resp. it should be split: Reset the fail timer for `(err != nil || !success)`, and inside that condition and before resting the timer have another if condition to increase the pause.

Also I think `f.pullPause` is now only used in this loop/serve function, so can be removed from the struct too.